src/execution/service.ts

```diff
@@ -876,6 +966,17 @@ export class ServiceScriptExecution extends BaseExecutionWithCommand<ServiceScri
         break;
       }
       case 'starting': {
+        this._state.readyMonitor?.abort();
```

Comment:
Nit: Does `this._state.child.kill` not need to be called here?

Reply:
Nope, the way it works that if we haven't started yet, then we go to the `stopping` state, and then in `_onChildStarted` we notice that we are stopped and kill the child. (We can't kill the child before it has started)